Frontend/src/pages/registration_edit/components/registration_editor.jsx

```diff
@@ -0,0 +1,122 @@
+import { EventSelector } from '@thewca/wca-components'
+import React, { useEffect, useState } from 'react'
+import { Button, Checkbox, TextArea } from 'semantic-ui-react'
+import getCompetitionInfo from '../../../api/competition/get/get_competition_info'
+import { getSingleRegistration } from '../../../api/registration/get/get_registrations'
+import { updateRegistration } from '../../../api/registration/patch/update_registration'
+import getCompetitorInfo from '../../../api/user/get/get_user_info'
+import { setMessage } from '../../../ui/events/messages'
+import LoadingMessage from '../../../ui/loadingMessage'
+import styles from './editor.module.scss'
+
+export default function RegistrationEditor({ user_id, competition_id }) {
+  const [competitorInfo, setCompetitorInfo] = useState({})
+  const [heldEvents, setHeldEvents] = useState([])
+  const [registration, setRegistration] = useState({})
+  const [comment, setComment] = useState('')
+  const [status, setStatus] = useState('')
+  const [selectedEvents, setSelectedEvents] = useState([])
+
+  useEffect(() => {
+    getCompetitorInfo(user_id).then((competitorInfo) =>
+      setCompetitorInfo(competitorInfo)
+    )
+  }, [user_id])
+  useEffect(() => {
+    getSingleRegistration(user_id, competition_id).then((response) => {
+      const registration = response.registration
+      setRegistration(registration)
+      setComment(registration.comment)
+      setStatus(registration.registration_status)
+      setSelectedEvents(registration.event_ids)
+    })
+  }, [user_id, competition_id])
+
+  useEffect(() => {
+    getCompetitionInfo(competition_id).then((competitionInfo) => {
+      setHeldEvents(competitionInfo.event_ids)
+    })
+  }, [competition_id])
+
+  return (
+    <div className={styles.editor}>
+      {!registration.registration_status ||
```

Comment:
It's a bit inconsistent that here we check whether something is loading by seeing if the data exists yet, but other components we have an explicit `isLoading` state. Personally I prefer the former (less state), but either is fine. Consistency would be nice.
There's also the potential edge-case of a competition having 0 events (before it's been setup). Seems unlikely someone would need this page in that case, but technically the loading UI would be displayed forever.

Reply:
This page is not usable/reachable when the competition hasn't been setup, because it requires a registration to be present. I agree with the consistency argument, I will remove the isLoading states throughout in a different PR that introduces a global loading component